xwiki-commons-core/xwiki-commons-properties/src/main/java/org/xwiki/properties/internal/DefaultBeanDescriptor.java

```diff
@@ -286,6 +286,22 @@ private void setCommonProperties(DefaultPropertyDescriptor desc, Map<Class, Anno
             }
             group.setFeature(parameterFeature.value());
         }
+
+
+        PropertyDisplayType displayTypeAnnotation = (PropertyDisplayType) annotations.get(PropertyDisplayType.class);
+        Type displayType;
+        if (displayTypeAnnotation != null && displayTypeAnnotation.value().length > 0) {
+            Class[] types = displayTypeAnnotation.value().clone();
+            ArrayUtils.reverse(types);
+            displayType = types[0];
+
+            for (int i = 1; i < types.length; i++) {
+                displayType = new DefaultParameterizedType(null, types[i], displayType);
```

Comment:
Using a `for` does not make any sense to me. With that code it's impossible to have several parameters, like Map<String, Integer> for example.

Reply:
Right, I was thinking of having just a hierarchy of classes. Do you think we should restrict the type to be of the form `Class1<Class2, Class3, ...>`?

Reply:
> Do you think we should restrict the type to be of the form Class1<Class2, Class3, ...>

That's what I meant with 99%, 1% is stuff like `Map<String, List<Integer>>` I think. If there is really a string need for it we can think about it later, not easy to express in an annotation.